### PR TITLE
Strict encoding

### PR DESCRIPTION
**Pull request by zkao**
@h4sh3d Please see the usage of strict encoding required by the node

monero strict encoding strategy implemented here: 
https://github.com/zkao/rust-lnpbp/tree/farcaster_encoding

### CODE REVIEW

File: src/role.rs [hunk 1]
```diff
@@ -47,15 +48,13 @@ pub enum BlockchainRole {
...
+    /// Defines the type of timelock used for the arbitrating transactions
     type Timelock: Copy;
+    /// Defines the address format for the arbitrating blockchain
+    type Address: StrictEncode + StrictDecode;
 }
-
 /// An accordant is the blockchain which does not need transaction inside the protocol nor
 /// timelocks, it is the blockchain with the less requirements for an atomic swap.
-pub trait Accordant: Blockchain + Keys + Curve + Commitment {}
+pub trait Accordant: Blockchain + Keys + Curve + Commitment + Clone + PrivateViewKey {}
```
Comment (h4sh3d) [inline src/role.rs:60]: `PrivateViewKey` I don't like this as it is too specific to Monero. This should be fixed latter.

File: src/protocol_message.rs [hunk 2]
```diff
@@ -1,17 +1,30 @@
...
-use crate::crypto::Proof;
+use crate::crypto::{Commitment, Proof, Signatures, Curve};
 use crate::role::{Accordant, Arbitrating};
 
 /// Trait for defining inter-daemon communication messages.
 pub trait ProtocolMessage {}
 
+use strict_encoding::{StrictDecode, StrictEncode};
+
+pub trait ProtMsg {}
```
Comment (h4sh3d) [inline src/protocol_message.rs:11]: Why `ProtMsg` if we already have `ProtocolMessage`?

File: src/role.rs (hunk 1 above)
Comment (zkao) [reply src/role.rs:60]: view is a field of the generic `RevealAliceSessionParams` protocol message, could be something like accessory data
``` rust
    /// The `K_s^a` spend public key
    pub view: Ac::PrivateViewKey,
```

File: src/protocol_message.rs (hunk 2 above)
Comment (zkao) [reply src/protocol_message.rs]: fixed in 0b95fcc

File: src/role.rs (hunk 1 above)
Comment (h4sh3d) [reply src/role.rs:60]: Yes, this is already non-generic. I propose we keep it as it is now and we do latter a PR that apply genericness everywhere.

**Pull request merged**